iOS/New/Views/Common/Settings/WebView.swift

```diff
@@ -57,16 +62,32 @@ struct WebView: UIViewRepresentable {
         }
 
         func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
-            Task {
+            Task { @MainActor in
+                if let sourceId = parent.sourceId {
+                    let localStorage = await webView.getLocalStorage()
+                    for (key, value) in localStorage {
+                        UserDefaults.standard.set(value, forKey: "\(sourceId).\(key)")
```

Comment:
instead of passing in the source id and assigning to the user defaults here, it would be more ideal to do the same thing as cookies and pass in a binding for the local storage keys, then assign them in the same place as the cookies are. it would be better if we could namespace them with the login settings key like cookies.